discovery_engine_core/core/src/engine.rs

```diff
@@ -318,7 +322,7 @@ where
             stack.data = StackData::default();
         }
         update_stacks(
-            stacks.values_mut(),
+            &mut *stacks,
```

Comment:
🎨 we can remove the dereference operator here (and also on the other places where we have `&mut *stacks`)
```suggestion
            &mut stacks,
```